### PR TITLE
feat: Set up Spend Permission Utilities + fetchPermissions

### PR DESCRIPTION
**Pull request by fan-zhang-sv**
### Summary
This PR introduces public utility functions for managing spend permissions, starting with the `fetchPermissions` method. This provides developers with a convenient way to query existing spend permissions for account-spender pairs on specific chains.

#### 🔧 **New Public API**
- **Added new export**: `./spend-permission` entry point in `package.json` 
- **New utility function**: `fetchPermissions()` for querying existing spend permissions
- **Comprehensive test coverage**: Full test suite with edge cases and error handling

#### 📁 **File Changes**
- `packages/account-sdk/package.json` - Added new export entry for spend permission utilities
- `packages/account-sdk/src/interface/public-utilities/spend-permission/index.ts` - Main export file
- `packages/account-sdk/src/interface/public-utilities/spend-permission/methods/fetchPermissions.ts` - Implementation
- `packages/account-sdk/src/interface/public-utilities/spend-permission/methods/fetchPermissions.test.ts` - Test suite

#### `fetchPermissions()`
- **Purpose**: Retrieves all spend permissions granted by a specific account to a spender on a given chain
- **Use Cases**: 
  - Check existing permissions before creating new ones
  - Display current allowances to users
  - Audit spend permission usage
- **Parameters**: 
  - `account`: The account address that granted permissions
  - `chainId`: The chain ID (automatically converted to hex format)
  - `spender`: The spender address that received permissions
- **Returns**: Array of `SpendPermission` objects

#### Usage Example
```typescript
import { fetchPermissions } from '@base-org/account/spend-permission';

const permissions = await fetchPermissions({
  account: '0x1234...',
  spender: '0x5678...',
  chainId: 8453 // Base mainnet
}, provider);

console.log(`Found ${permissions.length} permissions`);
```

### Testing
- ✅ Successful permission fetching
- ✅ Empty result handling
- ✅ Chain ID format conversion (decimal to hex)
- ✅ Error propagation and RPC error handling
- ✅ Multiple test scenarios with different chain IDs

### Breaking Changes
None - this is a purely additive change that introduces new public utilities.

### Next Steps
This establishes the foundation for additional spend permission utilities. Future PRs may include:
- Permission creation utilities
- Permission revocation helpers
- Permission validation tools

### _How did you test your changes?_

Screenshot 2025-07-31 at 2 18 20 PM

### CODE REVIEW

File: packages/account-sdk/src/interface/public-utilities/spend-permission/methods/fetchPermissions.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * console.log(`Found ${permissions.length} permissions`);
+ * permissions.forEach(permission => {
+ *   console.log(`Token: ${permission.permission.token}`);
+ * });
+ * ```
+ */
+export const fetchPermissions = async (
+  request: FetchPermissionsType,
+  provider: ProviderInterface
+): Promise<SpendPermission[]> => {
```
Comment (stephancill) [inline packages/account-sdk/src/interface/public-utilities/spend-permission/methods/fetchPermissions.ts:48]: how do we feel about changing this to take one argument: `FetchPermissionsType & { provider: ProviderInterface }`?

**Review by stephancill — APPROVED** [review]
